### PR TITLE
Media: Manage media scale through Redux preferences state

### PR DESCRIPTION
**Pull request by aduth**
Related: #8270

This pull request seeks to migrate the media scale preference from the legacy Flux implementation to the newer Redux-based preferences state. In doing so, it introduces a new standalone component for rendering the scale control, which is an app component controlling the preference value. This value is injected directly into the `<MediaLibraryList />` component where it is used.

`<MediaLibrary />` is one of two components still using `<PreferencesData />`, the other being `<PollInvitation />` slated to be removed in #8270. If #8270 is merged before this, I will rebase and remove the `<PreferencesData />` component.

__Testing Instructions:__

Verify that there are no regressions in behavior of the [post editor](http://calypso.localhost:3000/post) media library scale control at both desktop and mobile viewport sizes. Notably, you should confirm that the media scale allows smooth scrolling between predefined scale options and that a network request to save the preference is triggered after a one second delay. Upon refreshing, your preference should be persisted.

Ensure that Mocha tests pass:

```
npm run test-client
```

cc @deBhal @nylen

### CODE REVIEW

**Comment by deBhal** [comment]
#8270 Is merged and deployed, so you're clear to drop `PreferencesData`

**Comment by aduth** [comment]
>#8270 Is merged and deployed, so you're clear to drop `PreferencesData`

Rebased and removed. Thanks!

File: client/my-sites/media-library/scale.jsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import SegmentedControlItem from 'components/segmented-control/item';
+import { setPreference, savePreference } from 'state/preferences/actions';
+import { getPreference } from 'state/preferences/selectors';
+
+/**
+ * Constants
+ */
+const SLIDER_STEPS = 100;
+const SCALE_TOUCH_GRID = 0.32;
+const SCALE_CHOICES = [ 0.077, 0.115, 0.157, 0.24, 0.323 ];
```
Comment (nylen) [inline client/my-sites/media-library/scale.jsx]: This seems a bit "magical" on its own - I would add a comment similar to what was there before:

```
// Choices are 12, 8, 6, 4, and 3 items per row, with some horizontal padding between items
```

**Comment by aduth** [comment]
Thanks for the review @nylen . I added documentation for the constants in the rebased 3c8b4c2.

**Pull request merged**